app/models/energy/ScheduleRuleset.py

```diff
@@ -0,0 +1,207 @@
+"""Schedule Ruleset Schema"""
+from pydantic import BaseModel, Schema, validator, ValidationError, UrlStr, ConstrainedStr
+from typing import List, Union
+from enum import Enum
+from uuid import UUID, uuid4
+from datetime import datetime
+
+
+class DateSpecificationType(str, Enum):
+
+    date_range = 'DateRange'
+    specific_date = 'SpecificDate'
+
+
+class ScheduleContinuous(BaseModel):
+    """This Numeric Type allows all numbers, including fractional amounts, within the range to
+  be valid."""
+
+    schedule_continuous: float = Schema(
+        ...,
+    )
+
+
+class ScheduleDiscrete(BaseModel):
+    """This Numeric Type allows all only integers within the range to be valid."""
+
+    schedule_discrete: int = Schema(
+        ...,
+    )
+
+
+class NumericType (BaseModel):
+    """Designates how the range values are validated."""
+
+    numeric_type:  Union[ScheduleContinuous, ScheduleDiscrete]
+
+
+class UnitType (str, Enum):
+    dimensionless: 'Dimensionless'
+    temperature: 'Temperature'
+    delta_temperature: 'DeltaTemperature'
+    precipitation_rate: 'PrecipitationRate'
+    angle: 'Angle'
+    convection_coefficient: 'ConvectionCoefficient'
+    activity_level: 'ActivityLevel'
+    velocity: 'Velocity'
+    capacity: 'Capacity'
+    power: 'Power'
+    availability: 'Availability'
+    percent: 'Percent'
+    control: 'Control'
+    mode: 'Mode'
+
+
+class ScheduleTypeLimits (BaseModel):
+    """Specifies the data types and limits for values contained in schedules."""
```

Comment:
Are you going to implement custom schema validators which enforce these limits?

Reply:
Should I add custom validators that ensure `unit_type` such as `Temperature`, `ActivityLevel` have the correct `numeric_type` -`continuous` is assigned etc.?

Reply:
That seems good to me, @chriswmackey do you think that is overkill?

Reply:
I agree that I don't think that we need custom validators and it would be overkill